### PR TITLE
Refactor SupportDNNL functions

### PR DESCRIPTION
**Pull request by agrabows**
## Description ##
For any oneDNN primitive use for MXNet operator it is necessary to check if operators' inputs, parameters and sometimes outputs are compatible with primitive's boundaries. Almost every operator had its SupportDNNL function that was checking those conditions. Goal of this PR is to systematize those functions by making universal SupportDNNL<>() function with multiple use cases possible

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [x] Create universal SupportDNNL function
- [x] Refactor SupportDNNLOp functions using new function

### CODE REVIEW

**Comment by agrabows** [comment]
@mxnet-bot run ci [sanity]

**Comment by agrabows** [comment]
@mxnet-bot run ci [all]

**Comment by agrabows** [comment]
@mxnet-bot run ci [centos-cpu, centos-gpu, unix-gpu]

**Comment by agrabows** [comment]
@mxnet-bot run ci [unix-cpu, centos-gpu, unix-gpu]

File: src/operator/nn/dnnl/dnnl_softmax_output.cc [hunk 1]
```diff
@@ -90,9 +90,10 @@ static DNNLSoftmaxOutputFwd& GetSoftmaxOutputForward(const SoftmaxOutputParam& p
   return it->second;
 }
 
-//  This is only used for forward. For backward ,need double check compatibility
-bool SupportDNNLSoftmaxOutput(const SoftmaxOutputParam& param) {
-  return param.multi_output ? false : true;
+//  This is only used for forward. For backward one needs to need double check compatibility.
```
Comment (bgawrych) [inline src/operator/nn/dnnl/dnnl_softmax_output.cc:93]: two spaces at the beggining, second phrase is strange for me

**Review by bgawrych — APPROVED** [review]

File: src/operator/nn/dnnl/dnnl_softmax_output.cc (hunk 1 above)
Comment (bgawrych) [reply src/operator/nn/dnnl/dnnl_softmax_output.cc:93]:
```suggestion
```

**Comment by bgawrych** [comment]
@mxnet-bot run ci [clang]

**Pull request merged**